lib/puppet/pops/lookup/lookup_adapter.rb

```diff
@@ -393,7 +393,7 @@ def initialize_module_provider(lookup_invocation, module_name)
         mp.config = HieraConfig.v4_function_config(Pathname(mod.path), "#{module_name}::data", mp)
         mp
       else
-        raise Puppet::Error.new("Environment '#{environment.name}', cannot find module_data_provider '#{provider_name}'")
+        raise Puppet::Error.new(_("Environment '%{env}', cannot find module_data_provider '%{provider}'")) % { env: environment.name, provider: provider_name }
```

Comment:
@Magisus module_data_provider is a keyword, and should be left out of translation